src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -1787,11 +1814,57 @@ public static void setInterDCStreamThroughputOutboundMegabitsPerSec(int value)
         conf.inter_dc_stream_throughput_outbound_megabits_per_sec = value;
     }
 
-    public static String[] getAllDataFileLocations()
+    /**
+     * Checks if the local system data must be stored in a specific location which support redundancy.
+     *
+     * @return {@code true} if the local system keyspaces data must be stored in a different location,
+     * {@code false} otherwise.
+     */
+    public static boolean useSpecificLocationForSystemData()
+    {
+        return conf.system_data_file_directory != null;
+    }
+
+    /**
+     * Returns the locations where should be stored the local system keyspaces data.
+     *
+     * <p>If the {@code system_data_file_directory} was unspecified, the local system keyspaces data should be stored
+     * in the first data directory. This approach guaranty that the server can tolerate the lost of n - 1 disks.</p>
+     *
+     * @return the locations where should be stored the local system keyspaces data
+     */
+    public static String[] getSystemKeyspacesDataFileLocations()
+    {
+        if (conf.system_data_file_directory != null)
+            return new String[] {conf.system_data_file_directory};
+
+        return conf.data_file_directories.length == 0  ? conf.data_file_directories
+                                                       : new String[] {conf.data_file_directories[0]};
+    }
+
+    /**
+     * Returns the locations where should be stored the non local system keyspaces data.
+     *
+     * @return the locations where should be stored the non local system keyspaces data
+     */
+    public static String[] getNonSystemKeyspacesDataFileLocations()
```

Comment:
I was thinking that what determines if a table goes into the sepearate directory is the fact that it uses `LocalStrategy`, so we want it safer in a single directory, possibly with redundancy. AFAIK this strategy is only used by system tables, but if somehow there were not-system keyspaces with that strategy we would want them in the safer directory, I guess. So, perhaps we could call this `getLocalKeyspacesDataFileLocations`, and the property in cassandra.yaml `local_data_file_directory`, etc.? Does it make any sense?

Reply:
I understand your logic. It just seems to me that the `system` part is important because it implies that your server relies on it. I would be more in favor of `getLocalSystemKeyspacesDataFileLocations`.
Let's wait for the opinion of the other reviewer then I will change it to what we agree upon at that point.